crates/core/src/kernel/snapshot/iterators.rs

```diff
@@ -158,12 +157,12 @@ impl LogicalFileView {
     }
 
     /// Returns the raw JSON statistics string for this file, if available.
-    pub fn stats(&self) -> Option<&str> {
-        get_string_value(
-            self.files
-                .column(*FIELD_INDICES.get(FIELD_NAME_STATS).unwrap()),
-            self.index,
-        )
+    pub fn stats(&self) -> Option<String> {
+        let stats = self.stats_parsed()?.slice(self.index, 1);
+        let value = to_json(&stats)
+            .ok()
+            .map(|arr| arr.as_string::<i32>().value(0).to_string());
+        value.and_then(|v| (!v.is_empty()).then_some(v))
```

Comment:
we now need to serialise individual fields to get the json stats (for add actions) since we are no longer stacking the stats column.